app/assets/javascripts/host_edit.js

```diff
@@ -167,12 +164,11 @@ function submit_with_all_params() {
       $('#' + resource + '-progress').hide();
       $('#content').replaceWith($('#content', response));
       $(document.body).trigger('ContentLoad');
-      if ($('[data-history-url]').exists()) {
-        history.pushState(
-          {},
-          capitalized_resource + ' show',
-          $('[data-history-url]').data('history-url')
-        );
+      if (resource == 'host' && $("[data-history-url]").exists()) {
+        url_to_redirect = $("[data-history-url]").data('history-url');
+      }
+      if(url_to_redirect){
```

Comment:
I believe we can loose this `if` statement and join it safely with the previous one.
So basically keeping the original version only adding `resource == 'host' && ` in the `if` statement.